### ISSUE TITLE
Stop bot from choosing threefold repetition moves if it has advantage against opponents

### ISSUE THREAD

**Issue by segahen** [PanagiotisIatrou/chess-auto-bot#17]
Like title says, is there a way we can force the bot to choose another move in order stop it from drawing by threefold rep? i got matches where i was like +8 and above and still lose by threefold rep moves

**Comment by xHeaven** [comment]
I'll look into this today, I think I can fix this issue. I'll keep you updated.

**Comment by segahen** [comment]
@xHeaven thank you, nice to see you as a new contributor :D Also sometime when im in mouseless mode in the middle of the auto, the bot stops and i have to start again, i will send the log here later if it happens again

**Comment by PanagiotisIatrou** [comment]
> @xHeaven thank you, nice to see you as a new contributor :D Also sometime when im in mouseless mode in the middle of the auto, the bot stops and i have to start again, i will send the log here later if it happens again

Hello @segahen, I think the second issue you mentioned is a duplicate of https://github.com/PanagiotisIatrou/chess-auto-bot/issues/10. Do you also notice the illegal move errors on the console when the bot stops?

**Comment by xHeaven** [comment]
Opened a PR to solve the Threefold Repetition issue: https://github.com/PanagiotisIatrou/chess-auto-bot/pull/18
Let me know when you can update us about the logs related to the bot stopping.

**Comment by segahen** [comment]
the threefold problem is solved, im happy with it. About the bug, i havent got it today so i dont know yet but i will post once i see

**Comment by xHeaven** [comment]
> the threefold problem is solved, im happy with it. About the bug, i havent got it today so i dont know yet but i will post once i see

Have you solved it with the PR I made or you found another approach?
As for the bug, thanks in advance.

**Comment by segahen** [comment]
Yes i used your PR and solve it, now it only draw with insuffi materials, which is perfect to me.

**Comment by PanagiotisIatrou** [comment]
Hello @segahen. Could you maybe share a game or two when this happened? Perhaps a link or the PGN. I think it would really help to investigate more on this issue.

**Comment by segahen** [comment]
@PanagiotisIatrou Heaven made a 3 fold check so the problem fixed :) no worry about this

**Comment by segahen** [comment]
1. e4 e6 2. Nc3 c5 3. Nf3 a6 4. d4 cxd4 5. Nxd4 b5 6. Bd3 Bb7 7. O-O Nc6 8. Nxc6
Bxc6 9. Re1 Bc5 10. Qh5 Bd4 11. Ne2 g6 12. Qh3 Bg7 13. c3 d6 14. Nd4 Bd7 15. Qg3
Qc7 $6 16. Nf5 Bf8 17. Ne3 Be7 18. Nc2 h6 19. Be3 Nf6 20. Rad1 e5 21. f3 $6 Be6
22. Nb4 Kf8 23. Bc2 a5 24. Nd5 $6 Bxd5 $1 25. exd5 Nh5 26. Qf2 Bg5 $1 27. a4 Bxe3 28.
Rxe3 Qb8 29. g3 Kg7 30. f4 exf4 $2 31. Rf3 Qa7 32. Qxa7 Rxa7 33. axb5 fxg3 34.
hxg3 Re8 35. Bd3 Nf6 36. c4 Nd7 37. Bf1 Nc5 38. Ra3 $6 Raa8 39. Kf2 a4 40. Re1 $6
Rxe1 $2 41. Kxe1 f5 42. Bd3 Rc8 43. Bc2 Kf6 $1 44. Kd2 $1 g5 45. Kc3 $4 g4 $2 46. Ra1 $1
f4 $1 47. gxf4 h5 $6 48. Rg1 $2 Rg8 49. Kd2 $2 Rh8 $2 50. Rg2 $2 Nb3+ 51. Ke3 Re8+ $2 52.
Kd3 $4 Rg8 $1 53. Bd1 Nc5+ 54. Kd4 Rg7 55. Rg1 Kf5 56. Rh1 Kxf4 57. Rxh5 g3 $1 58.
Rh4+ Kf5 $1 59. Rh5+ Kf6 60. Rh6+ Ke7 61. Rh1 g2 $1 62. Rg1 Rg3 $1 63. Bc2 Kd7 64.
Bf5+ Kc7 65. Bh7 Nb3+ 66. Ke4 Rg7 67. Bf5 Nd2+ $1 68. Kf4 Nxc4 69. Bg4 Rf7+ 70.
Kg3 Ne3 71. Bf3 Kb6 72. Kf2 Nc4 73. Ke2 Re7+ 74. Kd3 Ne5+ 75. Kc3 Nxf3 76. Rxg2
Kxb5 77. Rg8 Rc7+ 78. Kd3 Ne5+ 79. Kd2 Nc4+ 80. Kc2 Nb6+ 81. Kb1 Rh7 82. Ka2 Rh2
83. Rb8 Kc5 84. Ka3 Rh3+ 85. Ka2 Nc4 86. Rb7 Re3 87. Rb8 Kxd5 88. Rb5+ Kd4 89.
Rb4 Re2 90. Ka1 Re1+ 91. Ka2 Re2 92. Ka1 Re1+ 93. Ka2 Re2 1/2-1/2

Here is a good example that i encounterd today, i was up 4 points and could have easily trade the rook with the enemy but the bot or stockfish decided to repeat the same moves and ended up 3fold rep.

**Comment by segahen** [comment]
Here is a good example that i encounterd today, i was up 4 points and could have easily trade the rook with the enemy but the bot or stockfish decided to repeat the same moves and ended up 3fold rep. @PanagiotisIatrou 
[3fold.txt](https://github.com/PanagiotisIatrou/chess-auto-bot/files/10568153/3fold.txt)
